src/archivist/services/index.js

```diff
@@ -4,261 +4,284 @@ import { pathToFileURL } from 'url';
 
 import config from 'config';
 
+import * as exposedFilters from '../extract/exposedFilters.js';
+
 import Service from './service.js';
 import SourceDocument from './sourceDocument.js';
 import Terms from './terms.js';
 
 export const DECLARATIONS_PATH = './declarations';
 const declarationsPath = path.resolve(process.cwd(), config.get('@opentermsarchive/engine.collectionPath'), DECLARATIONS_PATH);
 
-export async function load(servicesIdsToLoad = []) {
-  let servicesIds = await getDeclaredServicesIds();
-
-  if (servicesIdsToLoad.length) {
-    servicesIds = servicesIds.filter(serviceId => serviceId.match(new RegExp(`^${servicesIdsToLoad.join('|')}$`, 'g')));
-  }
+const JSON_EXT = '.json';
+const JS_EXT = '.js';
+const HISTORY_SUFFIX = '.history';
+const FILTERS_SUFFIX = '.filters';
 
+export async function load(servicesIdsToLoad = []) {
+  const allServicesIds = await getDeclaredServicesIds();
+  const servicesIds = servicesIdsToLoad.length ? allServicesIds.filter(serviceId => servicesIdsToLoad.includes(serviceId)) : allServicesIds;
   const services = {};
 
   await Promise.all(servicesIds.map(async serviceId => {
-    const { name, terms } = await loadServiceDeclaration(serviceId);
+    services[serviceId] = await createServiceFromDeclaration(serviceId);
+  }));
 
-    const service = new Service({ id: serviceId, name });
+  return services;
+}
 
-    await Promise.all(Object.keys(terms).map(termsType => loadServiceDocument(service, termsType, terms[termsType])));
+export async function createServiceFromDeclaration(serviceId) {
+  const { name, terms: termsDeclarations } = await loadServiceDeclaration(serviceId);
+  const service = new Service({ id: serviceId, name });
 
-    services[serviceId] = service;
+  await Promise.all(Object.entries(termsDeclarations).map(async ([ termsType, termsDeclaration ]) => {
+    const sourceDocuments = await createSourceDocuments(service.id, termsDeclaration);
+
+    service.addTerms(new Terms({ service, type: termsType, sourceDocuments }));
   }));
 
-  return services;
+  return service;
 }
 
-async function loadServiceDeclaration(serviceId) {
-  const jsonDeclarationFilePath = path.join(declarationsPath, `${serviceId}.json`);
-  const rawServiceDeclaration = await fs.readFile(jsonDeclarationFilePath);
+export async function loadServiceDeclaration(serviceId) {
+  const filePath = path.join(declarationsPath, `${serviceId}${JSON_EXT}`);
 
   try {
-    return JSON.parse(rawServiceDeclaration);
+    return JSON.parse(await fs.readFile(filePath));
   } catch (error) {
-    throw new Error(`The "${serviceId}" service declaration is malformed and cannot be parsed in ${jsonDeclarationFilePath}`);
+    throw new Error(`The "${serviceId}" service declaration is malformed and cannot be parsed in ${filePath}`);
```

Comment:
This is not true anymore: with the change above now including file read in the `try` block, there are other potential failure reasons than a malformed file.